File: render-javafx/src/main/java/dev/kkorolyov/pancake/render/javafx/JavaFXText.java
package dev.kkorolyov.pancake.render.javafx;

import dev.kkorolyov.pancake.platform.media.graphic.RenderTransform;
import dev.kkorolyov.pancake.platform.media.graphic.shape.Text;

/**
 * {@link Text} implemented through JavaFX.
 */
public class JavaFXText extends Text {
	private final EnhancedGraphicsContext g;

	/**
	 * Constructs a new JavaFX-backed text instance.
	 * @param g backing graphics context
	 */
	public JavaFXText(EnhancedGraphicsContext g) {
		this.g = g;
	}

	@Override
	public void render(RenderTransform transform) {
		g
				.shape(this)
				.rotate(transform)
				.get()
				.strokeText(getValue(), transform.getPosition().getX(), transform.getPosition().getY());
	}
}

File: svc-app-render-jfx/src/main/kotlin/dev/kkorolyov/pancake/javafx/render/JavaFxText.kt
package dev.kkorolyov.pancake.javafx.render

import dev.kkorolyov.pancake.platform.media.graphic.RenderTransform
import dev.kkorolyov.pancake.platform.media.graphic.shape.Text

/**
 * [Text] implemented through JavaFX.
 */
class JavaFxText(private val g: EnhancedGraphicsContext) : Text() {
	override fun render(transform: RenderTransform) {
		g.run {
			shape(this@JavaFxText)
			rotate(transform)

			get().strokeText(value, transform.position.x, transform.position.y)
		}
	}
}
